core/src/main/java/org/infinispan/interceptors/distribution/L1LastChanceInterceptor.java

```diff
@@ -76,40 +76,50 @@ public void start() {
    }
 
    public CompletableFuture<Void> visitDataWriteCommand(InvocationContext ctx, DataWriteCommand command, boolean assumeOriginKeptEntryInL1) throws Throwable {
-      Object returnValue = ctx.forkInvocationSync(command);
-      Object key;
-      if (shouldUpdateOnWriteCommand(command) && command.isSuccessful() &&
-            cdl.localNodeIsOwner((key = command.getKey()))) {
-         if (trace) {
-            log.trace("Sending additional invalidation for requestors if necessary.");
+      ctx.onReturn((ctx1, command1, rv, throwable) -> {
+         if (throwable != null)
+            throw throwable;
+
+         Object key;
+         if (shouldUpdateOnWriteCommand(command) && command.isSuccessful() &&
+               cdl.localNodeIsOwner((key = command.getKey()))) {
+            if (trace) {
+               log.trace("Sending additional invalidation for requestors if necessary.");
+            }
+            // Send out a last attempt L1 invalidation in case if someone cached the L1
+            // value after they already received an invalidation
+            blockOnL1FutureIfNeeded(l1Manager
```

Comment:
It is, this PR is still not doing anything to make the interceptor chain actually stop blocking.